docs/user-guide/kubectl-conventions.md

```diff
@@ -8,15 +8,45 @@ assignees:
 * TOC
 {:toc}
 
-## Using `kubectl` in Reusable Scripts 
+## Using `kubectl` in Reusable Scripts
 
 If you need stable output in a script, you should:
 
-* Request one of the machine-oriented output forms, such as `-o name`, `-o json`, `-o yaml`, `-o go-template`, or `-o jsonpath` 
+* Request one of the machine-oriented output forms, such as `-o name`, `-o json`, `-o yaml`, `-o go-template`, or `-o jsonpath`
 * Specify `--output-version`, since those output forms (other than `-o name`) output the resource using a particular API version
 * Specify `--generator` to pin to a specific behavior forever, if using generator-based commands (such as `kubectl run` or `kubectl expose`)
 * Don't rely on context, preferences, or other implicit state
 
+## Generators
```

Comment:
Suggest nesting it under Best Practices > kubectl run > Generators (or `kubectl run --generators`, or both)